### PR TITLE
Migrate Brave Ads confirmations.json/issuers state from JSON to prefs

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/24041

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Confirm issuers are fetched and persisted (including after restart). Issuers are now persisted in `Default/Preferences`.

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_util.cc [hunk 1]
```diff
@@ -50,15 +51,20 @@ void SetIssuers(const IssuersInfo& issuers) {
...
 IssuersInfo GetIssuers() {
   IssuersInfo issuers;
+
   issuers.ping =
       AdsClientHelper::GetInstance()->GetIntegerPref(prefs::kIssuerPing);
-  issuers.issuers = ConfirmationStateManager::GetInstance()->GetIssuers();
+
+  const absl::optional<base::Value::List> list =
+      AdsClientHelper::GetInstance()->GetListPref(prefs::kIssuers);
+  DCHECK(list);
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_util.cc:66]: I would prefer to do if statement here as prefs could be changed outside.

File: vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_value_util.cc [hunk 2]
```diff
@@ -119,38 +121,36 @@ base::Value::List IssuerListToValue(const IssuerList& issuers) {
...
     if (!item.is_dict()) {
-      return absl::nullopt;
+      continue;
     }
 
     const base::Value::Dict& dict = item.GetDict();
-    const absl::optional<IssuerType>& type_optional = ParseIssuerType(dict);
-    if (!type_optional) {
-      return absl::nullopt;
+    const absl::optional<IssuerType>& type = ParseIssuerType(dict);
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_value_util.cc:140]: nit: reference is not needed here

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_value_util.cc:140]: Fixed

File: vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_util.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_util.cc:66]: +1 fixed

File: vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_util.cc [hunk 3]
```diff
@@ -108,16 +124,18 @@ absl::optional<IssuerInfo> GetIssuerForType(const IssuersInfo& issuers,
...
-  const IssuersInfo& issuers = GetIssuers();
+  const absl::optional<IssuersInfo> issuers = GetIssuers();
+  if (!issuers) {
+    return false;
+  }
 
-  const absl::optional<IssuerInfo>& issuer_optional =
-      GetIssuerForType(issuers, issuer_type);
-  if (!issuer_optional) {
+  const absl::optional<IssuerInfo>& issuer =
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_util.cc:132]: nit: reference is not needed

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/account/issuers/issuers_util.cc:132]: Fixed. Also fixed other refs for issuers

**Review by aseren — APPROVED** [review]

**Pull request merged**